### PR TITLE
refactor: support angular 16

### PR DESCRIPTION
**Pull request by splincode**

### CODE REVIEW

File: projects/core/components/scrollbar/scrollbar.component.ts [hunk 1]
```diff
@@ -59,6 +59,6 @@ export class TuiScrollbarComponent {
         const scrollTop = offsetTop + offsetHeight / 2 - clientHeight / 2;
         const scrollLeft = offsetLeft + offsetWidth / 2 - clientWidth / 2;
 
-        nativeElement.scrollTo(scrollLeft, scrollTop);
+        nativeElement.scrollTo?.(scrollLeft, scrollTop);
```
Comment (splincode) [inline projects/core/components/scrollbar/scrollbar.component.ts:62]: I don't know why `Jest` fail in `combo-box.component.spec.ts` without that

File: package.json [hunk 2]
```diff
@@ -97,45 +97,60 @@
...
+        "@angular/router": "16.2.12",
+        "@ng-web-apis/universal": "3.0.7",
+        "@nguniversal/builders": "16.2.0",
+        "@nguniversal/express-engine": "16.2.0",
         "@nx/angular": "17.3.1",
         "@nx/jest": "17.3.1",
         "@nx/workspace": "17.3.1",
-        "@schematics/angular": "15.2.10",
+        "@schematics/angular": "16.2.12",
+        "@stackblitz/sdk": "1.9.0",
```
Comment (nsbarsukov) [inline package.json:123]: This deps should be only inside demo's `package.json`:
https://github.com/taiga-family/taiga-ui/blob/8915881b55d50c0da9894fe9b0e3c43ee016ccd1/projects/demo/package.json#L17

File: package.json [hunk 3]
```diff
@@ -97,45 +97,60 @@
...
+        "@angular/compiler-cli": "16.2.12",
+        "@angular/core": "16.2.12",
+        "@angular/forms": "16.2.12",
+        "@angular/localize": "16.2.12",
+        "@angular/platform-browser": "16.2.12",
+        "@angular/platform-browser-dynamic": "16.2.12",
+        "@angular/platform-server": "16.2.12",
+        "@angular/router": "16.2.12",
+        "@ng-web-apis/universal": "3.0.7",
+        "@nguniversal/builders": "16.2.0",
```
Comment (nsbarsukov) [inline package.json:117]: https://github.com/taiga-family/taiga-ui/blob/8915881b55d50c0da9894fe9b0e3c43ee016ccd1/projects/demo/package.json#L24

File: package.json [hunk 4]
```diff
@@ -97,45 +97,60 @@
...
+        "@nguniversal/express-engine": "16.2.0",
```
Comment (nsbarsukov) [inline package.json:118]: https://github.com/taiga-family/taiga-ui/blob/8915881b55d50c0da9894fe9b0e3c43ee016ccd1/projects/demo/package.json#L16

File: package.json [hunk 5]
```diff
@@ -97,45 +97,60 @@
...
-        "@angular/animations": "15.2.10",
-        "@angular/cdk": "15.2.9",
-        "@angular/cli": "15.2.10",
-        "@angular/common": "15.2.10",
-        "@angular/compiler-cli": "15.2.10",
-        "@angular/core": "15.2.10",
-        "@angular/forms": "15.2.10",
-        "@angular/platform-browser": "15.2.10",
-        "@angular/router": "15.2.10",
+        "@angular-builders/custom-webpack": "16.0.1",
```
Comment (nsbarsukov) [inline package.json:100]: https://github.com/taiga-family/taiga-ui/blob/8915881b55d50c0da9894fe9b0e3c43ee016ccd1/projects/demo/package.json#L23

Comment (nsbarsukov) [reply package.json:100]: Remove all extra deps.

And run 
```shell
rm -rf package-lock.json node_modules **/node_modules
npm i --workspaces --include-workspace-root
```

File: package.json [hunk 6]
```diff
@@ -97,45 +97,60 @@
...
+        "@schematics/angular": "16.2.12",
+        "@stackblitz/sdk": "1.9.0",
         "@taiga-ui/browserslist-config": "0.6.0",
         "@taiga-ui/commitlint-config": "0.7.1",
         "@taiga-ui/cspell-config": "0.36.0",
         "@taiga-ui/eslint-plugin-experience": "0.65.0",
         "@taiga-ui/prettier-config": "0.10.0",
         "@taiga-ui/stylelint-config": "0.20.0",
         "@taiga-ui/tsconfig": "0.17.0",
+        "@tinkoff/ng-dompurify": "4.0.0",
```
Comment (nsbarsukov) [inline package.json:131]: https://github.com/taiga-family/taiga-ui/blob/8915881b55d50c0da9894fe9b0e3c43ee016ccd1/projects/demo/package.json#L18

File: package.json [hunk 7]
```diff
@@ -97,45 +97,60 @@
...
+        "date-fns": "3.3.1",
+        "express": "4.18.3",
         "http-server": "14.1.1",
         "husky": "9.0.11",
         "kill-port": "2.0.1",
         "lint-staged": "15.2.2",
-        "ng-packagr": "15.2.2",
+        "ng-packagr": "16.2.3",
         "nx": "17.2.8",
+        "parse5": "6.0.1",
```
Comment (nsbarsukov) [inline package.json:146]: https://github.com/taiga-family/taiga-ui/blob/8915881b55d50c0da9894fe9b0e3c43ee016ccd1/projects/cdk/package.json#L24

**Review by nsbarsukov — CHANGES_REQUESTED** [review]

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**